src/tv2-common/videoSwitchers/TriCaster.ts

```diff
@@ -163,6 +187,108 @@ export class TriCaster extends VideoSwitcherImpl {
 		throw new Error('Method not implemented.')
 	}
 
+	private generateDveBoxLayers(boxes: any[]): Partial<Record<TSR.TriCasterLayerName, TSR.TriCasterLayer>> {
+		return {
+			a: boxes[0].enabled ? this.generateDveBoxLayout(boxes[0]) : this.generateInvisibleBoxLayer(),
+			b: boxes[1].enabled ? this.generateDveBoxLayout(boxes[1]) : this.generateInvisibleBoxLayer(),
+			c: boxes[2].enabled ? this.generateDveBoxLayout(boxes[2]) : this.generateInvisibleBoxLayer(),
+			d: boxes[3].enabled ? this.generateDveBoxLayout(boxes[3]) : this.generateInvisibleBoxLayer()
+		}
+	}
+
+	private generateDveBoxLayout(box: any): TSR.TriCasterLayer {
+		return {
+			input: `${SOURCE_INPUT_PREFIX}${box.source}`,
+			positioningAndCropEnabled: true,
+			position: this.convertPosition(box),
+			scale: this.convertScale(box),
+			crop: this.convertCrop(box)
+		}
+	}
+
+	private convertPosition(box: any): TSR.TriCasterLayer['position'] {
+		return {
+			x: this.convertPositionX(box),
+			y: this.convertPositionY(box)
+		}
+	}
+
+	/**
+	 * The x-position needs an offset added depending on the scale.
+	 * It seems to be a linear progression that for every 48 'size', that we are from 100% scale, an offset of 0.009 needs to be added.
+	 * If the x is negative we need to subtract the offset instead of adding it.
+	 * The division by a 1000 is to convert from the value we receive to the value TriCaster accepts
+	 */
+	private convertPositionX(box: { x: number; size: number }): number {
+		// 1000 comes ATEM scale upper bound. 48 comes from ATEM x position upper bound divided by 100
+		const offset = ((1000 - box.size) / 48) * 0.009
+		return box.x / 1000 + (box.x < 0 ? offset * -1 : offset)
+	}
+
+	/**
+	 * The y-position needs an offset that depend on the percentage y is compared to the height of the screen
+	 * If y is negative we need to subtract the offset instead of adding it.
+	 * The division by a 1000 is to convert from the value we receive to the value TriCaster accepts
+	 */
+	private convertPositionY(box: { y: number }): number {
+		const positiveValue = box.y < 0 ? box.y * -1 : box.y
+		const percentageToBeAdded = (positiveValue / 2700) * 100
+		const offset = (percentageToBeAdded * positiveValue) / 100
+		const position = box.y < 0 ? box.y - offset : box.y + offset
+		return (position / 1000) * -1
```

Comment:
I think this can be simplified to `return box.y*(1+Math.abs(box.y)/2700)/1000`

Reply:
I agree my current implementation needs a refactor.
I would like us to hit the sweetspot where it's not oneline magic, but still not as verbose as it currently is :)